tools/Custom/JsonExtensions.cs

```diff
@@ -155,8 +155,9 @@ private static void ProcessBody(JToken token)
                         try
                         {
                             JToken parsedValue = JToken.Parse(stringValue);
-                            property.Value = parsedValue; // Replace with unescaped JSON object
-                            ProcessBody(parsedValue); // Recursively process
+                            string originalToken = JsonConvert.SerializeObject(parsedValue, Formatting.None, new PreserveStringConverter()); // Ensures that the value matches the original type
```

Comment:
`value.Type` is asserted to be string at this time(line 152).
I'm pretty sure serializing the value isn't going to change it. Plus you won't be encountering a number at this point.

Reply:
On a separate note, line 158 should probably be removed. parsing the string to a JToken is really unncessary given that we already have the `value` variable which is already a JToken.